app/scripts/modules/core/account/account.service.ts

```diff
@@ -182,8 +182,8 @@ export class AccountService {
         const available: string[] = intersection(all, this.cloudProviderRegistry.listRegisteredProviders());
         let result: string[];
         if (application) {
-          if (application.attributes.cloudProviders) {
-            result = application.attributes.cloudProviders.split(',');
+          if (application.attributes.cloudProviders.length) {
```

Comment:
Correct - we now ensure `accounts` and `cloudProviders` are arrays via the `applicationReader`.